.github/workflows/publish_to_hex_pm.yml

```diff
@@ -33,11 +29,26 @@ jobs:
         with:
           ref: ${{ github.event.workflow_run.head_branch || github.ref }}
 
-      - name: Set up Elixir ${{ env.ELIXIR_VERSION }} and OTP ${{ env.OTP_VERSION }}
-        uses: erlef/setup-elixir@v1
+      - name: Setup asdf
+        uses: asdf-vm/actions/setup@v1
+
+      - name: Cache asdf
+        uses: actions/cache@v2
```

Comment:
They published `actions/cache@v3`, should we change to use the latest version instead?
And should we take this chance to update `actions/checkout` to v3 as well?
If you agree, please also change the other files 🙏